apps/server/src/shared/repo/federalstate/federalstate.repo.integration.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { NotFoundError, NullCacheAdapter } from '@mikro-orm/core';
+import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
+import { Test, TestingModule } from '@nestjs/testing';
+import { FederalState } from '@shared/domain';
+import { MongoMemoryDatabaseModule } from '@shared/infra/database';
+import { cleanupCollections, federalStateFactory } from '@shared/testing';
+import { FederalStateRepo } from './federalstate.repo';
+
+describe('federalstate repo', () => {
+	let module: TestingModule;
+	let repo: FederalStateRepo;
+	let em: EntityManager;
+
+	beforeAll(async () => {
+		// em.clear do not clear the resultCache, it must be disabled for this test
+		module = await Test.createTestingModule({
+			imports: [MongoMemoryDatabaseModule.forRoot({ resultCache: { adapter: NullCacheAdapter } })],
+			providers: [FederalStateRepo],
+		}).compile();
+		repo = module.get(FederalStateRepo);
+		em = module.get(EntityManager);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	afterEach(async () => {
+		await em.nativeDelete(FederalState, {});
+		em.clear();
+		await cleanupCollections(em);
+	});
+
+	it('should be defined', () => {
+		expect(repo).toBeDefined();
+		expect(typeof repo.findById).toEqual('function');
+	});
+
+	it('should implement entityName getter', () => {
+		expect(repo.entityName).toBe(FederalState);
+	});
+
+	describe('findById', () => {
+		it('should return right keys', async () => {
+			const fsA = federalStateFactory.build();
```

Comment:
Please keep in mind to use our test structure. With describe senarios and boundled setup method. :D

Reply:
done